diff --git a/BUILD.bazel b/BUILD.bazel
index 8547feb..f736dce 100755
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -152,11 +152,11 @@ autoconf(
     ] + select({
         ":bazel_m4_path": [
             # Prevent any host installed m4 from being run.
-            checks.AC_DEFINE_UNQUOTED("M4", "\"/bin/false\""),
+            checks.AC_DEFINE("M4", "\"/bin/false\""),
         ],
         "//conditions:default": [
             # Allow users to have a canonical flex binary that would look for host m4.
-            checks.AC_DEFINE_UNQUOTED("M4", "\"m4\""),
+            checks.AC_DEFINE("M4", "\"m4\""),
         ],
     }),
     deps = [":package"],
