diff --git a/BUILD.bazel b/BUILD.bazel
index 00fc3f3..ea2a090 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -2,6 +2,7 @@ load("@bazel_skylib//rules:write_file.bzl", "write_file")
 load("@rules_cc//cc:cc_binary.bzl", "cc_binary")
 load("@rules_cc//cc:cc_library.bzl", "cc_library")
 load("@rules_cc_autoconf//autoconf:autoconf.bzl", "autoconf")
 load("@rules_cc_autoconf//autoconf:autoconf_hdr.bzl", "autoconf_hdr")
+load("@rules_cc_autoconf//autoconf:autoconf_linkopts.bzl", "autoconf_linkopts")
 load("@rules_cc_autoconf//autoconf:autoconf_srcs.bzl", "autoconf_srcs")
 load("@rules_cc_autoconf//autoconf:checks.bzl", "checks")
@@ -1022,6 +1023,12 @@ cc_library(
 # src/ -> bin/patch
 # ---------------------------------------------------------------------------
 
+autoconf_linkopts(
+    name = "getrandom_linkopts",
+    vars = ["GETRANDOM_LIB"],
+    deps = [":configure_ac"],
+)
+
 cc_binary(
     name = "patch",
     srcs = [
@@ -1039,8 +1046,7 @@ cc_binary(
         "src",
     ],
     linkopts = select({
         "@platforms//os:linux": ["-lpthread"],
-        "@rules_cc//cc/compiler:msvc-cl": ["advapi32.lib"],
         "//conditions:default": [],
     }),
     local_defines = LOCAL_DEFINES + [
@@ -1048,6 +1054,7 @@ cc_binary(
     ],
     visibility = ["//visibility:public"],
     deps = [
+        ":getrandom_linkopts",
         ":libpatch",
     ],
 )
